packages/credential-store/src/types/ICredentialStore.ts

```diff
@@ -0,0 +1,121 @@
+
+export interface ICredentialStore extends IPluginMethodMap {
+  /**
+   * Add a new credential.
+   * @param args
+   */
+  crsAddCredential(args: AddCredentialArgs): Promise<DigitalCredential>
+
+  /**
+   * Update credential the state of an existing credential.
+   * @param args
+   */
+  crsUpdateCredentialState(args: UpdateCredentialStateArgs): Promise<DigitalCredential>
+
+  /**
+   * Get a single credentials by primary key
+   * @param args
+   */
+  crsGetCredential(args: GetCredentialArgs): Promise<DigitalCredential>
+
+  /**
+   * Find one or more credentials using filters
+   * @param args
+   */
+  crsGetCredentials(args: GetCredentialsArgs): Promise<Array<DigitalCredential>>
+
+  /**
+   * Find one or more credentials using filters
+   * @param args
+   */
+  crsGetUniqueCredentials(args: GetCredentialsArgs): Promise<Array<UniqueDigitalCredential>>
+
+  /**
+   * Find one credential by id or hash
+   * @param CredentialRole
+   * @param idOrHash
+   */
+  crsGetUniqueCredentialByIdOrHash(args: GetCredentialsByIdOrHashArgs): Promise<OptionalUniqueDigitalCredential>
+
+  /**
+   * Returns a list of UniqueDigitalCredentials that match the given filter based on the claims they contain.
+   * @param args
+   */
+  crsGetCredentialsByClaims(args: GetCredentialsByClaimsArgs): Promise<Array<UniqueDigitalCredential>>
+
+  /**
+   * Returns a count of UniqueDigitalCredentials that match the given filter based on the claims they contain.
+   * @param args
+   */
+  crsGetCredentialsByClaimsCount(args: GetCredentialsByClaimsArgs): Promise<number>
+
+  /**
+   * Delete a single credentials by primary key
+   * @param args
+   */
+  crsDeleteCredential(args: DeleteCredentialArgs): Promise<boolean>
+
+  /**
+   * Delete multiple credentials records using filters
+   * @param args
+   */
+  crsDeleteCredentials(args: DeleteCredentialsArgs): Promise<number>
+}
+
+export type GetCredentialArgs = {
+  id: string
+}
+
+export type GetCredentialsArgs = {
+  filter: FindDigitalCredentialArgs
+}
+
+export type GetCredentialsByClaimsArgs = {
+  filter: FindClaimsArgs
+  credentialRole?: CredentialRole
+  tenantId?: string
+}
+
+export type GetCredentialsByIdOrHashArgs = {
+  credentialRole: CredentialRole
+  idOrHash: string
+}
+
+export type DeleteCredentialArgs = {
+  id: string
+}
+
+export type DeleteCredentialsArgs = GetCredentialsArgs
+
+export type AddDigitalCredential = Omit<
+  NonPersistedDigitalCredential,
+  'id' | 'documentType' | 'documentFormat' | 'uniformDocument' | 'hash' | 'createdAt' | 'lastUpdatedAt' | 'validFrom' | 'validUntil'
+>
+
+export type AddCredentialArgs = {
+  credential: AddDigitalCredential
+}
+
+export type { UpdateCredentialStateArgs } from '@sphereon/ssi-sdk.data-store' // TODO create a local copy?
+
+export interface UniqueDigitalCredential {
+  hash: string
+  digitalCredential: DigitalCredential
```

Comment:
I would also expect the an optional id value in here. For (V)C this is a mandatory attribute that is unique in the scope of an issuer (so not perse unique in a DB)

Reply:
fixed in next commit